### DISCUSSION TITLE
YouTube Playlists

### DISCUSSION THREAD

**Discussion by mrdschrute** [jmbannon/ytdl-sub#355] [General]
First, thanks for the work you've put into this project. I'm wondering if you have thoughts on automating channel playlists for use in Plex. I mainly watch educational videos, so viewing them by playlist (topic) often makes the most sense. Finding a subject when the videos are organized by year is tough. I see the [documentation](https://ytdl-sub.readthedocs.io/en/latest/presets.html?highlight=plex) about collections, but that seems to describe adding each playlist manually. Of course, if you are downloading videos regularly, playlist could change (new video added) and new playlist will be created. As you know, yt-dlp can download all of the playlists from a channel and put them in there own folder, but then they are not optimized for Plex. Do you have any ideas on the best approach? I have some basic Python skills, so I'm happy to help with coding and testing. If you don't think that feature is a good fit for this project, that's fine also.

**Comment by jmbannon** [comment]
Are you referring to super-playlists (i.e. a playlist of playlists). This should totally be doable with the current ytdl-sub version but might involve some config tinkering. Can you provide an example URL you'd want to scrape?

EDIT: GitHub Issue for this: https://github.com/jmbannon/ytdl-sub/issues/502

**Comment by mrdschrute** [comment]
I'm thinking of channels like CrashCourse [https://www.youtube.com/@crashcourse/playlists](url). The channel has a bunch of videos, and organizing them by date doesn't make much sense. They should really be organized by playlist/subject. I see that it can be done with "collection_season_1" and "collection_season_1_url", but that channel alone has 50+ created playlists and occasionally adds new ones. I wondering if there is a way to automate "download the videos in created playlists > make each playlist a season within that channel for Plex (unless there is a better approach) > periodically update for new videos in existing playlists or new playlists. I think I could probably work this out with yt-dlp and a bash script to rename/organize, but I thought it might be useful to others and your work on the project might have given you some insight into the best approach. Again, if it's not currently possible and you don't think it's a good fit for the project, that's totally fine. On the other hand, if it would be useful, I would be willing to help code/test.

Thanks

**Reply by jmbannon** [reply]
ytdl-sub should support putting these in nicely named folders, but I see a few issues regarding Plex/Jellyfin support:

1. Both do not support named seasons via metadata scraping
  - Would be very hard to know which playlist a season belongs to - will only show Season 1, Season 2, etc
2. There's no reproducible way to map a playlist to a season number
  - i.e. "Back to School" to season 1, "History" to season 2, etc
  - ytdl-sub tries its best to be stateless, would probably need to add some sort of playlist_id to index mapping, which I really want to avoid

For me, (1) is a deal-breaker to fix (2) because either way, it's not going to look good in the player.

**Reply by jmbannon** [reply]
One idea that'd be possible today is to treat each playlist as its own TV show, and set a shared genre.

**Reply by mrdschrute** [reply]
Thanks for getting back to me. Totally understand your explanation for why that functionality is not a good fit.

**Reply by 0Downtime** [reply]
I get that this is a hard feature to implement but hopefully it's something can be implemented in the future if Jellyfin or Plex makes such functionality easier to accommodate. Educational content on YouTube heavily uses playlists. I suppose it's possible to create playlists manually in Jellyfin, but having the functionality built into ytdl-sub would be amazing. I get the desire to stay stateless though. Thanks for the hard work on the project, it really has been an awesome tool! 

I am new to Jellyfin, so if someone has a suggestion on how I can have adhoc playlist url imports be put into their own category that isn't necessarily a season that might mitigate the need for built in playlist support.
